lib/src/realm_class.dart

```diff
@@ -656,3 +702,17 @@ class MigrationRealm extends DynamicRealm {
 
   MigrationRealm._(Realm realm) : super._(realm);
 }
+
+/// The signature of a callback that will be executed while the Realm is opened asynchronously with [Realm.open].
+/// This is the registered callback onProgressCallback to receive progress notifications while the download is in progress.
+///
+/// * syncProgress - an object of [SyncProgress] that contains `transferredBytes` and `transferableBytes`.
+/// {@category Realm}
+typedef ProgressCallback = void Function(SyncProgress syncProgress);
+
+/// @nodoc
+extension $CancellableCompleterExtension<T> on Completer<T> {
+  CancellableCompleter<T> asCancellable(CancellationToken? cancellationToken, {OnCancelCallback? onCancel}) {
+    return CancellableCompleter<T>(cancellationToken, onCancel: onCancel);
+  }
```

Comment:
This is not really an extension method, as no state from the original completer is used. I think we should drop it
```suggestion
```